pkg/helper/roles/helpers.go

```diff
@@ -43,6 +43,36 @@ func GetOperatorRoleName(cluster *cmv1.Cluster, missingOperator *cmv1.STSOperato
 	return awsCommonUtils.TruncateRoleName(role)
 }
 
+func CheckRedHatManagedTag(arn string, awsClient aws.Client, r *rosa.Runtime) {
+	roleName, err := aws.GetResourceIdFromARN(arn)
+	if err != nil {
+		r.Reporter.Errorf("Failed to get resource ID from ARN %s: %v", arn, err)
+		return
+	}
+
+	attachedPolicies, err := awsClient.GetPolicyDetailsFromRole(&roleName)
+	if err != nil {
+		r.Reporter.Errorf("Failed to get policy details for role '%s': %v", roleName, err)
+		return
+	}
+
+	hasRedHatManagedTag := false
+
+tagSearch:
+	for _, policy := range attachedPolicies {
+		for _, tag := range policy.Policy.Tags {
+			if *tag.Key == tags.RedHatManaged {
+				hasRedHatManagedTag = true
+				break tagSearch
+			}
+		}
+	}
+
+	if !hasRedHatManagedTag {
+		r.Reporter.Warnf("The role '%s' is not a Red Hat managed role", roleName)
+	}
```

Comment:
We don't need to check each policy- the role itself will suffice here. The role will have the `red-hat-managed` tag